src/components/searchBar/index.tsx

```diff
@@ -1,23 +1,97 @@
-import React, { useCallback, useState } from "react";
+import React, { useCallback, useState } from "react"
+
+interface IngredientProps {
+  id: number
+  name: string
+}
 
 export default function SearchBar() {
-  const [value, setValue] = useState("");
+  const [searchQuery, setSearchQuery] = useState("")
+  const [ingredients, setIngredients] = useState<IngredientProps[]>([])
+  const [selectedIngredients, setSelectedIngredients] = useState<
+    IngredientProps[]
+  >([])
 
   const handleChangeInput = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
-      event.preventDefault();
-      setValue(event.target.value);
+      event.preventDefault()
+      setSearchQuery(event.target.value)
+      fetch(
+        `https://api.spoonacular.com/food/ingredients/autocomplete?apiKey=${process.env.NEXT_PUBLIC_SPOONACULAR_API_KEY}&query=${event.target.value}&number=10&metaInformation=true`
+      )
+        .then((res) => res.json())
+        .then((result) => {
+          console.log(result)
+          setIngredients(result)
+        })
     },
     []
-  );
+  )
+
+  const handleSelectIngredient = useCallback(
+    (ingredient: IngredientProps) => {
+      if (
+        selectedIngredients.some(
+          (selected) => selected.name === ingredient.name
+        )
+      ) {
+        alert("Ingredient already selected. Please, choose another one.")
+        return
+      }
+      setSelectedIngredients((prev) => [
+        ...prev,
+        {
+          ...ingredient,
+        },
```

Comment:
Can we simplify like this?
```suggestion
      setSelectedIngredients((prev) => [
        ...prev,
        ingredient
```